codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java

```diff
@@ -362,6 +362,32 @@ public void addBodyAttribute(String name, String value) throws ErrorDataEncoderE
      */
     public void addBodyFileUpload(String name, File file, String contentType, boolean isText)
             throws ErrorDataEncoderException {
+       addBodyFileUpload(name, file.getName(), file, contentType, isText);
+    }
+
+    /**
+     * Add a file as a FileUpload
+     *
+     * @param name
+     *            the name of the parameter
+     * @param file
+     *            the file to be uploaded (if not Multipart mode, only the filename will be included)
+     * @param filename
+     *            the filename to use for this File part, empty String will be ignored by
+     *            the encoder
+     * @param contentType
+     *            the associated contentType for the File
+     * @param isText
+     *            True if this file should be transmitted in Text format (else binary)
+     * @throws NullPointerException
+     *             for name and file
+     * @throws ErrorDataEncoderException
+     *             if the encoding is in error or if the finalize were already done
+     */
+    public void addBodyFileUpload(String name, String filename, File file, String
+            contentType,
```

Comment:
I wonder we should remove the "filename" argument and just add a boolean that will tell that file.getName() should not be used but an empty one. WDYT @smaldini  ?